radix-engine/src/vm/wasm/wasmi.rs

```diff
@@ -910,6 +910,29 @@ fn secp256k1_ecdsa_verify_and_key_recover(
         .map(|buffer| buffer.0)
 }
 
+fn secp256k1_ecdsa_verify_and_key_recover_uncompressed(
+    mut caller: Caller<'_, HostState>,
+    message_ptr: u32,
+    message_len: u32,
+    signature_ptr: u32,
+    signature_len: u32,
+) -> Result<u64, InvokeError<WasmRuntimeError>> {
+    let runtime = grab_runtime!(caller);
+    let memory = grab_memory!(caller);
+
+    let message = read_memory(caller.as_context_mut(), memory, message_ptr, message_len)?;
+    let signature = read_memory(
+        caller.as_context_mut(),
+        memory,
+        signature_ptr,
+        signature_len,
+    )?;
+
+    runtime
+        .crypto_utils_secp256k1_ecdsa_verify_and_key_recover_uncompressed(message, signature)
+        .map(|buffer| buffer.0)
```

Comment:
what does this method do? returns `u64` and not a public key? why do we call `buffer.0`? should we not return the full buffer and update the return type to be `Result<[u8: 64], _>`?

Reply:
We can not pass the array raw to wasm as a function return, and have to wrap it into a buffer (which will be read by wasm later).

The wasmi linker requires a u32/u64 return type, so we had to unwrap the `Buffer`.

Reply:
so the caller of this function reads out the 65 bytes how? via `mut caller`? I lack the WASM knowledge to understand this :D but just looked wrong. I assume it is some WASM vodoo magic making this work...

Reply:
wrote simultaneously... thx makes sense